wasm/src/algorithms/bhp/mod.rs

```diff
@@ -125,4 +123,168 @@ mod tests {
+
+            let native_bits = PlaintextNative::Literal(native_literals[i].clone(), OnceCell::new()).to_bits_le();
+
+            let native_hash_256 = native_bhp256.hash(&native_bits).unwrap();
+            let native_hash_512 = native_bhp512.hash(&native_bits).unwrap();
+            let native_hash_768 = native_bhp768.hash(&native_bits).unwrap();
+            let native_hash_1024 = native_bhp1024.hash(&native_bits).unwrap();
+            let native_hash_to_group_256 = native_bhp256.hash_uncompressed(&native_bits).unwrap();
+            let native_hash_to_group_512 = native_bhp512.hash_uncompressed(&native_bits).unwrap();
+            let native_hash_to_group_768 = native_bhp768.hash_uncompressed(&native_bits).unwrap();
+            let native_hash_to_group_1024 = native_bhp1024.hash_uncompressed(&native_bits).unwrap();
+            let native_commit_256 = native_bhp256.commit(&native_bits, &scalar).unwrap();
+            let native_commit_512 = native_bhp512.commit(&native_bits, &scalar).unwrap();
+            let native_commit_768 = native_bhp768.commit(&native_bits, &scalar).unwrap();
+            let native_commit_1024 = native_bhp1024.commit(&native_bits, &scalar).unwrap();
+            let native_commit_to_group_256 = native_bhp256.commit_uncompressed(&native_bits, &scalar).unwrap();
+            let native_commit_to_group_512 = native_bhp512.commit_uncompressed(&native_bits, &scalar).unwrap();
+            let native_commit_to_group_768 = native_bhp768.commit_uncompressed(&native_bits, &scalar).unwrap();
+            let native_commit_to_group_1024 = native_bhp1024.commit_uncompressed(&native_bits, &scalar).unwrap();
+
+            assert_eq!(hash_256, Field::from(native_hash_256));
+            assert_eq!(hash_512, Field::from(native_hash_512));
+            assert_eq!(hash_768, Field::from(native_hash_768));
+            assert_eq!(hash_1024, Field::from(native_hash_1024));
+            assert_eq!(hash_to_group_256, Group::from(native_hash_to_group_256));
+            assert_eq!(hash_to_group_512, Group::from(native_hash_to_group_512));
+            assert_eq!(hash_to_group_768, Group::from(native_hash_to_group_768));
+            assert_eq!(hash_to_group_1024, Group::from(native_hash_to_group_1024));
+            assert_eq!(commit_256, Field::from(native_commit_256));
+            assert_eq!(commit_512, Field::from(native_commit_512));
+            assert_eq!(commit_768, Field::from(native_commit_768));
+            assert_eq!(commit_1024, Field::from(native_commit_1024));
+            assert_eq!(commit_to_group_256, Group::from(native_commit_to_group_256));
+            assert_eq!(commit_to_group_512, Group::from(native_commit_to_group_512));
+            assert_eq!(commit_to_group_768, Group::from(native_commit_to_group_768));
+            assert_eq!(commit_to_group_1024, Group::from(native_commit_to_group_1024));
+        }
+
+        let struct_pt: Plaintext<CurrentNetwork> = Plaintext::from_str(TEST_STRUCT).unwrap();
```

Comment:
Same here, this is the same type as `PlaintextNative`, use the `wasm` struct definition we've defined in the SDK instead.